h2o-core/src/main/java/water/fvec/persist/FramePersist.java

```diff
@@ -0,0 +1,111 @@
+package water.fvec.persist;
+
+import water.*;
+import water.fvec.Chunk;
+import water.fvec.Frame;
+import water.fvec.Vec;
+import water.util.FileUtils;
+
+import java.net.URI;
+
+import static water.fvec.persist.PersistUtils.*;
+
+public class FramePersist {
+    
+    static {
+        TypeMap.onIce(FrameMeta.class.getName());
+    }
+
+    private final Frame frame;
+
+    public FramePersist(Frame frame) {
+        this.frame = frame;
+    }
+    
+    private static class FrameMeta extends Iced<FrameMeta> {
+        Key<Frame> key;
+        String[] names;
+        Key<Vec>[] keys;
+        Vec[] vecs;
+        
+        FrameMeta(Frame f) {
+            key = f._key;
+            names = f.names();
+            keys = f.keys();
+            vecs = f.vecs();
+        }
+    }
+
+    private static URI getMetaUri(Key key, String dest) {
+        return FileUtils.getURI(dest + "/" + key);
+    }
+
+    private static URI getDataUri(String metaUri, int cidx) {
+        return FileUtils.getURI(metaUri + "_" + H2O.SELF.index() + "_" + cidx);
+    }
+
+    public void saveTo(String uri) {
+        uri = sanitizeUri(uri);
+        URI metaUri = getMetaUri(frame._key, uri);
+        write(metaUri, ab -> {
```

Comment:
it would be safer to first have the chunks loaded and after that load the frame envelope, otherwise something can start working with the partial frame before the chunks are actually available